command/server_dev.go

```diff
@@ -0,0 +1,828 @@
+
+	username string
+	password string
+}
+
+func enableQuickstart(ctx context.Context, init *vault.InitResult, core *vault.Core) (*quickstartOptions, error) {
+	options := &quickstartOptions{
+		idName:       "Lorem Ipsum",
+		idPolicyName: "developer",
+		idPolicy:     "path \"secret/*\" {capabilities = [\"create\", \"read\", \"update\", \"delete\"]}",
+
+		appRoleRoleName:   "app-read-only",
+		appRolePolicyName: "read-only",
+		appRolePolicy:     "path \"secret/*\" {capabilities = [\"read\"]}",
+
+		username: "lipsom",
+		password: "superSecretDemo",
+	}
+
+	// app role policy
+	r := &logical.Request{
+		Operation:   logical.UpdateOperation,
+		ClientToken: init.RootToken,
+		Path:        fmt.Sprintf("sys/policy/%s", options.appRolePolicyName),
+		Data: map[string]interface{}{
+			"name":   options.appRolePolicyName,
+			"policy": options.appRolePolicy,
+		},
+	}
+	resp, err := core.HandleRequest(ctx, r)
+	if err != nil {
+		return nil, fmt.Errorf("error creating policy: %w", err)
+	}
+	if resp.IsError() {
+		return nil, fmt.Errorf("failed to create policy: %w", resp.Error())
+	}
+
+	// userpass policy
+	r = &logical.Request{
+		Operation:   logical.UpdateOperation,
+		ClientToken: init.RootToken,
+		Path:        fmt.Sprintf("sys/policy/%s", options.idPolicyName),
+		Data: map[string]interface{}{
+			"name":   options.idPolicyName,
+			"policy": options.idPolicy,
+		},
+	}
+	resp, err = core.HandleRequest(ctx, r)
+	if err != nil {
+		return nil, fmt.Errorf("error creating policy: %w", err)
+	}
+	if resp.IsError() {
+		return nil, fmt.Errorf("failed to create policy: %w", resp.Error())
+	}
+
+	// userpass
+	r = &logical.Request{
+		Operation:   logical.UpdateOperation,
+		ClientToken: init.RootToken,
+		Path:        "sys/auth/userpass",
+		Data: map[string]interface{}{
+			"type":        "userpass",
+			"path":        "userpass/",
+			"description": "sample userpass method",
+		},
+	}
+	resp, err = core.HandleRequest(ctx, r)
+	if err != nil {
+		return nil, fmt.Errorf("error creating userpass: %w", err)
+	}
+	if resp.IsError() {
+		return nil, fmt.Errorf("failed to create userpass: %w", resp.Error())
+	}
+	r = &logical.Request{
+		Operation:   logical.UpdateOperation,
+		ClientToken: init.RootToken,
+		Path:        fmt.Sprintf("auth/userpass/users/%s", options.username),
+		Data: map[string]interface{}{
+			"password":       options.password,
+			"token_policies": []string{"default"},
+		},
+	}
+	resp, err = core.HandleRequest(ctx, r)
+	if err != nil {
+		return nil, fmt.Errorf("error creating userpass user: %w", err)
+	}
+	if resp.IsError() {
+		return nil, fmt.Errorf("failed to create userpass user: %w", resp.Error())
+	}
+
+	// identity
+	r = &logical.Request{
+		Operation:   logical.CreateOperation,
+		ClientToken: init.RootToken,
+		Path:        "identity/entity",
+		Data: map[string]interface{}{
+			"name": options.idName,
+			"metadata": map[string]string{
+				"organization": "ACME Inc.",
+				"team":         "Squad-2",
```

Comment:
At the moment I think it makes sense to keep the `quickstartOptions` struct as lean as possible, only sharing info that needs to be shown in the banner, since we won't be providing a method to manipulate these values. In the future, we can improve this UX but I don't want to paint ourselves into a corner.